src/pages/create-appointment/AppointmentAssignment.tsx

```diff
@@ -0,0 +1,188 @@
+import { LFMatch } from '../../types/lernfair/Match';
+import { LFSubCourse } from '../../types/lernfair/Course';
+import { getTrafficStatus } from '../../Utility';
+import { DateTime } from 'luxon';
+import CenterLoadingSpinner from '../../components/CenterLoadingSpinner';
+import { Assignment } from '../../types/lernfair/Appointment';
+
+type AssignmentProps = {
+    next: () => void;
+    back: () => void;
+};
+
+const query = gql`
+    query StudentCourseMatches {
+        me {
+            firstname
+            student {
+                matches {
+                    id
+                    uuid
+                    dissolved
+                    pupil {
+                        firstname
+                        lastname
+                        schooltype
+                        grade
+                        subjectsFormatted {
+                            name
+                        }
+                    }
+                    pupilEmail
+                }
+                subcoursesInstructing {
+                    id
+                    participantsCount
+                    maxParticipants
+                    published
+                    lectures {
+                        start
+                        duration
+                    }
+                    firstLecture {
+                        id
+                        start
+                        duration
+                    }
+                    course {
+                        name
+                        description
+                        tags {
+                            name
+                        }
+                    }
+                }
+            }
+        }
+    }
+`;
+
+const AppointmentAssignment: React.FC<AssignmentProps> = ({ next, back }) => {
+    const { data, loading } = useQuery(query);
+    const { t } = useTranslation();
+
+    const activeMatches = useMemo(() => data?.me?.student?.matches.filter((match: LFMatch) => !match.dissolved), [data?.me?.student?.matches]);
+
+    const publishedSubcourses = useMemo(
+        () => data?.me?.student?.subcoursesInstructing.filter((sub: LFSubCourse) => sub.published),
+        [data?.me?.student?.subcoursesInstructing]
+    );
+
+    const subcoursesToShow = useMemo(() => {
+        if (!publishedSubcourses) return [];
+
+        const sortedCourses: LFSubCourse[] = publishedSubcourses.sort((a: LFSubCourse, b: LFSubCourse) => {
+            const aLecture = a.firstLecture;
+            const bLecture = b.firstLecture;
+
+            if (!bLecture) return -1;
+            if (!aLecture) return 1;
+
+            const aDate = DateTime.fromISO(aLecture.start).toMillis();
+            const bDate = DateTime.fromISO(bLecture.start).toMillis();
+
+            if (aDate < DateTime.now().toMillis()) return 1;
+            if (bDate < DateTime.now().toMillis()) return 1;
+
+            if (aDate === bDate) return 0;
+            return aDate > bDate ? 1 : -1;
+        });
```

Comment:
I guess this should do the job honestly ☺️ 
```suggestion
        const sortedCourses: LFSubCourse[] = publishedSubcourses.sort((a: LFSubCourse, b: LFSubCourse) => {
            const aInMillis = DateTime.fromISO(a.firstLecture.start).toMillis();
            const bInMillis = DateTime.fromISO(b.firstLecture.start).toMillis();
            return aInMillis - bInMillis;
        });
```